src/sqlfluff/core/config.py

```diff
@@ -455,6 +455,7 @@ def __init__(
         ignore_local_config: bool = False,
         overrides: Optional[dict] = None,
         plugin_manager: Optional[pluggy.PluginManager] = None,
+        require_dialect: bool = True,
```

Comment:
By default, config requires a dialect to be set, but this can be overridden.

Reply:
Presumably this should only really be overridden for tests? Worth a comment?

Edit: Ah see further down it can also not be set at top level (my test case!). OK definitely think worthy of a comment.

```suggestion
        # Ideally a dialect should be set when config is read but sometimes
        # it might only be set in nested .sqlfluff config files, so allow it
        # to be not required.
        require_dialect: bool = True,
```